osu.Game.Tournament/TournamentGameBase.cs

```diff
@@ -223,15 +223,22 @@ private bool addBeatmaps()
 
             foreach (var r in ladder.Rounds)
             {
-                foreach (var b in r.Beatmaps)
+                foreach (var b in r.Beatmaps.ToList())
                 {
-                    if (b.BeatmapInfo == null && b.ID > 0)
+                    if (b.BeatmapInfo == null)
```

Comment:
Maybe invert this check and `continue`? Might make this a little bit easier to read. I did a double take when I saw this condition duplicated later down the line.